react/features/prejoin/middleware.js

```diff
@@ -0,0 +1,95 @@
+// @flow
+
+import {
+    ADD_PREJOIN_AUDIO_TRACK,
+    ADD_PREJOIN_VIDEO_TRACK,
+    PREJOIN_START_CONFERENCE
+} from './actionTypes';
+import { setPrejoinAudioMuted, setPrejoinVideoMuted } from './actions';
+import { SET_AUDIO_MUTED, SET_VIDEO_MUTED } from '../base/media';
+import { participantUpdated, getLocalParticipant } from '../base/participants';
+import { MiddlewareRegistry } from '../base/redux';
+import { updateSettings } from '../base/settings';
+import { getAllPrejoinConfiguredTracks, getPrejoinName } from './functions';
+
+declare var APP: Object;
+
+/**
+ * The redux middleware for {@link PrejoinPage}.
+ *
+ * @param {Store} store - The redux store.
+ * @returns {Function}
+ */
+MiddlewareRegistry.register(store => next => action => {
+    switch (action.type) {
+    case ADD_PREJOIN_AUDIO_TRACK: {
+        const { value: audioTrack } = action;
+
+        if (audioTrack) {
+            store.dispatch(
+                    updateSettings({
+                        micDeviceId: audioTrack.getDeviceId()
+                    }),
+            );
+        }
+
+        break;
+    }
+
+    case ADD_PREJOIN_VIDEO_TRACK: {
+        const { value: videoTrack } = action;
+
+        if (videoTrack) {
+            store.dispatch(
+                    updateSettings({
+                        cameraDeviceId: videoTrack.getDeviceId()
+                    }),
+            );
+        }
+
+        break;
+    }
+
+    case PREJOIN_START_CONFERENCE: {
+        const { dispatch, getState } = store;
+
+        _syncParticipantName(dispatch, getState);
+        APP.conference.prejoinStart(
+                getAllPrejoinConfiguredTracks(getState()),
```

Comment:
The tracks created in the prejoin feature don't have any constraints applied to them AFAICT. So we either make them disposable and create the desired tracks with contraints after, or we apply the contraints when the tracks are created.

Reply:
Sorry, but I don't quite understand. What constraints should I be aware of? 

That function only mutes/unmutes the tracks(according to the user's preferences). It follows the same flow from the `init` method.

Reply:
Do you mean I have to recreate the tracks with mute/unmute options? Can't I just do that on the fly? 🤔

Reply:
Ah, nevermind, I ddouble checked and the ccopnstraints (desired resolution and so on) are added internally. We are good!